### PR TITLE
feat: support extra plugin dir by EGG_EXTRA_PLUGIN_DIR

### PR DESCRIPTION
**Pull request by yugasun**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message follows commit guidelines

##### Description of change

`Optimize for serverless case using layer`

### CODE REVIEW

**Comment by fengmk2** [comment]
@atian25 Take a look at this feature request?

File: test/loader/mixin/load_plugin.test.js [hunk 1]
```diff
@@ -15,6 +15,8 @@ const EggLoader = require('../../..').EggLoader;
 describe('test/load_plugin.test.js', function() {
   let app;
 
+  process.env.EGG_EXTRA_PLUGIN_DIR = utils.getFilepath('outside-plugins');
```
Comment (fengmk2) [inline test/loader/mixin/load_plugin.test.js:18]: should use `mm` to mock env, e.g.:

```js
mm(process.env, 'EGG_EXTRA_PLUGIN_DIR', utils.getFilepath('outside-plugins'));
```

File: lib/loader/mixin/plugin.js [hunk 2]
```diff
@@ -348,6 +348,15 @@ module.exports = {
     // should find the $cwd/node_modules when test the plugins under npm3
     lookupDirs.push(path.join(process.cwd(), 'node_modules'));
 
+    // support extra plugin dir for serverless environment
+    // sometimes, developer will deploy node_modules to layer, when serverless function is starting up, node_modules is not the the same path as business code.
+    // so we need use process.env.EGG_EXTRA_PLUGIN_DIR to add lookupDirs for loading plugin.
+    // relative issue: https://github.com/serverless-components/tencent-egg/issues/11
+    const extraPluginDir = process.env.EGG_EXTRA_PLUGIN_DIR;
```
Comment (atian25) [inline lib/loader/mixin/plugin.js:355]: maybe we should also support multi dir

Comment (atian25) [reply lib/loader/mixin/plugin.js:355]: and it maybe better to support ENV for eggPath rather than pluginDir

**Pull request closed without merging**